python/glow/wgr/linear_model/tests/test_functions.py

```diff
@@ -85,48 +85,78 @@ def test_generate_alphas(spark):
     assert generate_alphas(df) == expected_alphas
 
 
+def test_labels_and_covars_ok(spark):
+    labeldf = pd.DataFrame({'Trait_1': [0, -1, 1], 'Trait_2': [0, 1, -1]})
+    covdf = pd.DataFrame({'Covariate_1': [0, 1, -1], 'Covariate_2': [0, -1, 1]})
+    with pytest.warns(None) as record:
+        validate_inputs(labeldf, covdf, 'continuous')
+
+        # Add a binary trait
+        labeldf['Trait_3'] = [0, 1, math.nan]
+        validate_inputs(labeldf, covdf, 'either')
+
+        # Delete continuous traits and check binary validation
+        del labeldf['Trait_1']
+        del labeldf['Trait_2']
+        validate_inputs(labeldf, covdf, 'binary')
+
+    assert len(record) == 0
+
+
 def test_assert_labels_all_present(spark):
-    labeldf = pd.DataFrame({'Trait_1': [-1, 1], 'Trait_2': [1, math.nan]})
-    covdf = pd.DataFrame({'Covariate_1': [1, -1], 'Covariate_2': [-1, 1]})
+    labeldf = pd.DataFrame({'Trait_1': [0, -1, 1], 'Trait_2': [0, 1, math.nan]})
+    covdf = pd.DataFrame({'Covariate_1': [0, 1, -1], 'Covariate_2': [0, -1, 1]})
     with pytest.raises(ValueError):
-        validate_inputs(labeldf, covdf)
+        validate_inputs(labeldf, covdf, 'continuous')
 
 
 def test_assert_covars_all_present(spark):
-    labeldf = pd.DataFrame({'Trait_1': [-1, 1], 'Trait_2': [1, -1]})
-    covdf = pd.DataFrame({'Covariate_1': [1, -1], 'Covariate_2': [-1, math.nan]})
+    labeldf = pd.DataFrame({'Trait_1': [0, -1, 1], 'Trait_2': [0, 1, -1]})
+    covdf = pd.DataFrame({'Covariate_1': [0, 1, -1], 'Covariate_2': [0, -1, math.nan]})
     with pytest.raises(ValueError):
         validate_inputs(labeldf, covdf)
 
 
 def test_check_labels_zero_mean(spark):
-    labeldf = pd.DataFrame({'Trait_1': [-1, 1], 'Trait_2': [1, 3]})
-    covdf = pd.DataFrame({'Covariate_1': [1, -1], 'Covariate_2': [-1, 1]})
+    labeldf = pd.DataFrame({'Trait_1': [0, -1, 1], 'Trait_2': [0, 1, 3]})
+    covdf = pd.DataFrame({'Covariate_1': [0, 1, -1], 'Covariate_2': [0, -1, 1]})
     with pytest.warns(UserWarning):
         validate_inputs(labeldf, covdf)
```

Comment:
Should these tests be with `binary` on?

Reply:
Oh sorry, I meant `continuous`. I wasn't sure if we should be testing with `either` given that this only holds for `continuous` and not `binary`.

Reply:
I can try both